pkg/featuregate/feature_gate.go

```diff
@@ -121,6 +122,8 @@ type MutableFeatureGate interface {
 	// overriding its default to true for a limited number of components without simultaneously
 	// changing its default for all consuming components.
 	OverrideDefault(name Feature, override bool) error
+	// SetLogger replaces the logger with the provided logger.
+	SetLogger(lg *zap.Logger)
```

Comment:
It's a little weird to add `SetLogger` into the interface. No any public interface methods depend on the zap.Logger.
Can we try to remove it?
```suggestion
```

Reply:
removed.

Reply:
Removing `SetLogger` is good, but on the other hand, it seems that you can't pass a valid `lg` to `featuregates.New`. Not a big problem though.